### PR TITLE
Adding support for project additional files

### PR DESCRIPTION
**Pull request by kwridan**
Part of https://github.com/tuist/tuist/issues/202

### Short description 📝

Often projects may include additional files or folder references that don't need to be part of any build phase (e.g. design references, documentation etc...)

### Solution 📦

`Workspace.swift` manifest now supports `additionalFiles`, the same concept can be added to the `Project.swift` manifest with the same rules.

- `.glob(pattern: "Documentation/**")`: glob pattern to one or more files
- `.folderReference(path: "Designs")`: single path to a directory that is added as a folder reference

Example:

```swift
let project = Project(name: "App",
                      targets: [
                           // ...
                      ],
                      additionalFiles: [
                         "Dangerfile.swift"
                         "Documentation/**",
                         .folderReference(path: "Designs")
                     ])
```

Screen Shot 2019-03-31 at 8 29 54 PM

### Implementation 👩‍💻👨‍💻

> Detail in a checklist the steps that you took to implement the PR.

- [x] Update `Project` manifest to include `additionalFiles`
- [x] Update models to translate globs to paths
- [x] Update `ProjectFileElements` to include `additionalFiles`
- [x] Include fixture
- [x] Update documentation
- [x] Update changelog

### Test Plan ⚒

- Run `tuist generate` within `fixtures/ios_app_with_custom_workspace/App`
- Open the generated workspace
- Verify `Dangerfile.swift` is included in the project but isn't part of the resource phase
- Verify `Documentation` files are included (with the appropriate group structure that reflects the file system) and that the files aren't copied to resources
- Verify `Server` folder reference is included

### CODE REVIEW

File: Sources/ProjectDescription/Project.swift [hunk 1]
```diff
@@ -6,19 +6,16 @@ public class Project: Codable {
     public let name: String
     public let targets: [Target]
     public let settings: Settings?
-
-    public enum CodingKeys: String, CodingKey {
-        case name
-        case targets
-        case settings
-    }
+    public let additionalFiles: [WorkspaceElement]
```
Comment (ollieatkinson) [inline Sources/ProjectDescription/Project.swift]: Is this still a workspace element?

Comment (pepibumur) [reply Sources/ProjectDescription/Project.swift]: If we'd like to reuse the same type I'd change the name.

File: fixtures/ios_app_with_custom_workspace/App/Server/server.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import json
```
Comment (pepibumur) [inline fixtures/ios_app_with_custom_workspace/App/Server/server.py]: 🤔

Comment (kwridan) [reply fixtures/ios_app_with_custom_workspace/App/Server/server.py]: heh - can replace this with a different fixture that doesn't look as dubious - the intentions was to show case a folder reference to some sever looking code

File: Sources/ProjectDescription/Project.swift (hunk 1 above)
Comment (pepibumur) [reply Sources/ProjectDescription/Project.swift]: I'd stick to the Xcode project's convention, [FileElement](https://github.com/tuist/xcodeproj/blob/master/Sources/xcodeproj/Objects/Files/PBXFileElement.swift#L5)

**Pull request merged**